app/controllers/api/comments_controller.rb

```diff
@@ -0,0 +1,20 @@
+class Api::CommentsController < ApplicationController
+  def create
+    post = Post.find(params[:post_id])
+    comment = Comment.new(author: @current_user, post:, text: param['text'])
+
+    if comment.save
+      render json: comment, status: 200
+    else
+      render json: {
+        error: 'Unable to create comment'
+      }
+    end
+  end
```

Comment:
- [ ] I am not able to create a comment because your application is demanding authentication. :thinking: It can be fixed if you make your controller something below :arrow_down:

![image](https://user-images.githubusercontent.com/93039359/216630205-3199297d-0ce3-4a3c-8785-c30c95cbf297.png)

Reply:
Hi @HAMMAS-SALEEM, the requirement says we should not set user by passing a parameter, that's why I use authentication to get the user.

Reply:
Hey Fayob, i hope you are doing well :blush: